.changeset/ten-rats-worry.md

```diff
@@ -0,0 +1,30 @@
+---
+'xstate': minor
+---
+
+You can now strongly type `meta` on state nodes and transitions via `stateMeta` and `transitionMeta` respectively:
+
+```ts
+import { createMachine } from 'xstate';
+
+const machine = createMachine({
+  types: {
+    stateMeta: {} as { title: string },
+    transitionMeta: {} as { button: 'primary' | 'secondary' }
```

Comment:
do we need this if it's not used by our runtime *anyhow*? One can only get access to this by introspecting the machine object - which is far from the most common use case (introducing a whole new generic for this purpose feels to me like a huge huge overcomplication).

On top of that, currently this type doesn't even flow to things like:
```ts
machine.getTransitionData(machine.getInitialState({} as any), { type: '' })[0].meta
```

Reply:
I actually think that the `.configuration` should be removed from the user-facing `state` 😅 If there is a need for introspection then a dedicated API should be introduced for that (even if that would be a getter~ on the `state` or smth). The current `state`'s shape is too fat and it actually creates problems for `@xstate/solid`